pkg/scheduler/ugm/manager.go

```diff
@@ -438,58 +475,100 @@ func (m *Manager) clearEarlierSetLimits(userLimits map[string]bool, groupLimits
 	for u, ut := range m.userTrackers {
 		// Is this user already tracked for the queue path?
 		if ut.IsQueuePathTrackedCompletely(queuePath) {
-			// Is there any limit config set for user in the current configuration? If not, then clear those old limit settings
-			if _, ok := userLimits[u]; !ok {
-				log.Log(log.SchedUGM).Debug("Need to clear earlier set configs for user",
-					zap.String("user", u),
-					zap.String("queue path", queuePath))
-				// Is there any running applications in end queue of this queue path? If not, then remove the linkage between end queue and its immediate parent
-				if ut.IsUnlinkRequired(queuePath) {
-					ut.UnlinkQT(queuePath)
-				} else {
-					ut.setLimits(queuePath, resources.NewResource(), 0)
-					log.Log(log.SchedUGM).Debug("Cleared earlier set limit configs for user",
-						zap.String("user", u),
-						zap.String("queue path", queuePath))
-				}
-				// Does "root" queue has any child queue trackers? At some point during this whole traversal, root might
-				// not have any child queue trackers. When the situation comes, remove the linkage between the user and
-				// its root queue tracker
-				if ut.canBeRemoved() {
-					delete(m.userTrackers, ut.userName)
+			// Traverse all the group trackers linked to user through different applications and remove the linkage
+			for appID, gt := range ut.appGroupTrackers {
+				if gt != nil {
+					g := gt.groupName
+					// Is there any limit config set for group in the current configuration? If not, then remove the linkage by setting it to nil
+					if ok := groupLimits[g]; !ok {
+						log.Log(log.SchedUGM).Debug("Removed the linkage between user and group through applications",
+							zap.String("user", u),
+							zap.String("group", gt.groupName),
+							zap.String("application id", appID),
+							zap.String("queue path", queuePath))
+						// removing the linkage only happens here by setting it to nil and deleting app id
+						// but group resource usage so far remains as it is because we don't have app id wise resource usage with in group as of now.
+						// YUNIKORN-1858 handles the group resource usage properly
+						// In case of only one (last) application, group tracker would be removed from the manager.
+						ut.setGroupForApp(appID, nil)
+						gt.removeApp(appID)
+						if len(gt.getTrackedApplications()) == 0 {
+							log.Log(log.SchedUGM).Debug("Is this app the only running application in group?",
```

Comment:
Nit: to me this is a weird sounding log message. The answer is a yes/no, but that's actually not logged. I suggest printing a statement here.

Reply:
fixing in the next step